pkg/option/config.go

```diff
@@ -950,6 +950,9 @@ const (
 	// ExternalClusterIPName is the name of the option to enable
 	// cluster external access to ClusterIP services.
 	ExternalClusterIPName = "bpf-lb-external-clusterip"
+
+	// BPFAllowVLAN instructs Cilium to allow vlan targeted packets
+	BPFAllowVLAN = "bpf-allow-vlan"
```

Comment:
I was thinking about `vlan-bpf-bypass` first, but current logic will malform vlan tagged packets and we need to drop or to bypass them to kernel. Think I need help in this question :)

Reply:
I think that technically there's two changes occurring in this PR:
a) Resolve the "malformed VLAN-tagged packets issue" by dropping packets with VLAN tags.
b) Provide a configuration option to bypass Cilium BPF logic.

(a) is basically a new default and I reason above[0] that this will just make it easier to detect when VLAN misconfiguration is causing issues in your network. (b) is then saying, if you know how the Linux stack is configured to handle VLAN traffic and you're OK with the implications, then you can teach Cilium to bypass its BPF logic for VLAN traffic.

I feel like longer term there is probably a more cohesive solution that involves teaching Cilium more about the overall VLAN configuration in your network and reasoning about the overall system + Cilium's part in it. But for now this combination of (a) / (b) should be enough to unblock what you're trying to do. And I'd argue that (b) is really what the configuration option is about.

[0]: https://github.com/cilium/cilium/pull/16772#discussion_r668158630

Reply:
Hm. It should be a good 'fix' for those who require vlan's. I will remove this complex logic and will make bpf drop or bypass all vlan tags depending on option with default to drop. Thanks for comments, will get back soon.